src/dhcp_mon.cpp

```diff
@@ -38,8 +38,28 @@ static struct event_mgr *main_event_mgr = NULL;
 static struct event_mgr *rx_event_mgr = NULL;
 /** libevent mgr struct */
 static struct event_mgr *tx_event_mgr = NULL;
+/** event to sync RX cache counter from COUNTERS_DB */
+struct event *ev_rx_cache_counter_update = NULL;
+/** event to sync RX cache counter from COUNTERS_DB */
+struct event *ev_tx_cache_counter_update = NULL;
 /** window_interval_sec monitoring window for dhcp relay health checks */
 static int db_update_interval_sec;
+/** When clearing counter is invoked, dhcpmon wouldn't write cache counter to COUNTERS_DB until it receives a signal,
+ *  in case recover signal is not sent by cli, add timeout 5s here. After 5s, dhcpmon would update COUNTERS_DB as previous.
+ */
+static constexpr int clear_counter_timeout = 5;
+/** Flag to determine whether to write cache counter to COUNTERS_DB.
+ *  0b11 - write (rx and tx cache counter are both up to date)
+ *  0b00 - don't write (Need to sync rx and tx cache counter from COUNTERS_DB)
+ *  0b01 - don't write (rx cache counter is up to date, but need to sync tx cache counter from COUNTERS_DB)
+ *  0b10 - don't write (tx cache counter is up to date, but need to sync rx cache counter from COUNTERS_DB) */
+static int write_counter_to_db = 0b11;
```

Comment:
How about define constants?
Then we can use the flag name in code instead of magic number.

```cpp
const int RX_READY = 0b01;
const int TX_READY = 0b10;

static int write_counter_to_db = RX_READY | TX_READY;
```

Reply:
Addressed